subprojects/core/src/main/java/org/gradle/internal/service/scopes/GlobalScopeServices.java

```diff
@@ -247,7 +250,8 @@ ObjectFactory createObjectFactory(InstantiatorFactory instantiatorFactory, Servi
             fileResolver,
             directoryFileTreeFactory,
             new DefaultFilePropertyFactory(fileResolver),
-            fileCollectionFactory);
+            fileCollectionFactory,
+            new DefaultDomainObjectCollectionFactory(instantiatorFactory, services, CollectionCallbackActionDecorator.NOOP, MutationGuards.identity()));
```

Comment:
Let's make this a service, so that other core things can use the internal factory.

Reply:
Done.